gradle-plugin/src/main/kotlin/com/github/buildeye/utils/SerializationUtils.kt

```diff
@@ -0,0 +1,48 @@
+package com.github.buildeye.utils
+
+import java.io.FileInputStream
+import java.io.FileOutputStream
+import java.io.ObjectInputStream
+import java.io.ObjectOutputStream
+import java.nio.file.Files
+import java.nio.file.NotDirectoryException
+import java.nio.file.Path
+
+fun <T> serialize(path: Path, obj: T?) {
+    ensureWritable(path)
+
+    FileOutputStream(path.toFile()).use { fileOutputStream ->
+        ObjectOutputStream(fileOutputStream).use { objectOutputStream ->
+            objectOutputStream.writeObject(obj)
+        }
+    }
+}
+
+fun <T> deserialize(path: Path): T {
+    FileInputStream(path.toFile()).use { fileInputStream ->
```

Comment:
You don't need to close `FileInputStream`, if you pass it to `ObjectInputStream`, and don't forget to close the latter.
```
ObjectInputStream(FileInputStream(path.toFile())).use { input -> ... }
```